src/Network/NWProtocolStack.cs

```diff
@@ -93,7 +120,7 @@ public void IterateProtocols (Action<NWProtocolOptions> callback)
 		public NWProtocolOptions InternetProtocol {
 			get {
 				var pHandle = nw_protocol_stack_copy_internet_protocol (GetCheckedHandle ());
-				return (pHandle == IntPtr.Zero) ? null : new NWProtocolOptions (pHandle, owns: true);
+				return (pHandle == IntPtr.Zero) ? null : new NWProtocolIPOptions (pHandle, owns: true);
```

Comment:
Maybe change the return type of the property to `NWProtocolIPOptions` (in a XAMCORE_4_0 if it's a breaking change)? Since that will always be the type of the returned object.

Reply:
It is a subclass, I don't think it would be a problem, the other way around will. API diff will let me know.